File: Bunga Gabungan/BungaGabungan.java
import java.util.Scanner;

public class BungaGabungan
{
    public static void main (String [] args)
    {
        Scanner scan = new Scanner (System.in);
        short P = scan.nextShort ();
        short Q = scan.nextShort ();

        scan.close ();

        int hasil = (P * P) + (Q * Q) + 1;
        System.out.print (hasil % 4 == 0 ? hasil : -1);
    }
}

File: Bunga Gabungan/Bunga-Gabungan.c
#include <stdio.h>

int main ()
{
    short P = 0, Q = 0;
    scanf ("%hd %hd", &P, &Q);

    int hasil = (P * P) + (Q * Q) + 1;
    hasil % 4 == 0 ? printf ("%d", hasil) : printf ("-1");

    return 0;
}